Formula/chapel.rb

```diff
@@ -23,19 +23,16 @@ def install
     ENV["CHPL_HOME"] = libexec
     # This is for mason
     ENV["CHPL_RE2"] = "bundled"
+    ENV["CHPL_GMP"] = "system"
+    ENV["CHPL_INCLUDE_PATH"] = "/usr/local/include"
+    ENV["CHPL_LIB_PATH"] = "/usr/local/lib"
 
     # Must be built from within CHPL_HOME to prevent build bugs.
     # https://github.com/Homebrew/legacy-homebrew/pull/35166
     cd libexec do
       system "./util/printchplenv", "--all"
       system "make"
-      # Need to let chapel choose target compiler with llvm
-      ENV["CHPL_HOST_CC"] = ENV["CC"]
-      ENV["CHPL_HOST_CXX"] = ENV["CXX"]
-      ENV.delete("CC")
-      ENV.delete("CXX")
-      system "./util/printchplenv", "--all"
-      system "make"
+      system "CHPL_LLVM=none make"
```

Comment:
```suggestion
      with_env(CHPL_LLVM: "none") do
        system "make"
      end
```
Similarly for the test block.